core/quorum_validation.go

```diff
@@ -167,3 +178,110 @@ func (c *Core) getUnpledgeId(wt string) string {
 	}
 	return b.GetUnpledgeId()
 }
+
+/*
+ * Function to check whether the TokenState is pinned or not
+ * Input tokenId, index, resultArray, waitgroup,quorumList
+ */
+func (c *Core) checkTokenState(tokenId, did string, index int, resultArray []TokenStateCheckResult, wg *sync.WaitGroup, quorumList []string) {
```

Comment:
Would it be a good option to enable this check at receiver side also? It's more a precautionary action by receiver to make sure the DHTprovs nodes are only quorum for its own transaction. In worst case situations if ever 2 consensus go through at same time , it can be detected at receiver level.